src/extensions/types.ts

```diff
@@ -247,14 +251,65 @@ export interface Param {
   advanced?: boolean;
 }
 
+export const isParam = (param: any): param is Param => {
+  return typeof param === "object" && Boolean(param.param && param.label);
+};
+
 export enum ParamType {
   STRING = "STRING",
   SELECT = "SELECT",
   MULTISELECT = "MULTISELECT",
+  SELECT_RESOURCE = "SELECT_RESOURCE",
   SECRET = "SECRET",
 }
 
 export interface ParamOption {
   value: string;
   label?: string;
 }
+
+// Typeguard for ExtensionSpec. (We often get "specs" from parsing yaml).
+// This helps decide if it's actually a spec or just some random yaml.
+// Any is reasonable here since this is a typeguard.
```

Comment:
Nit: I think this is a textbook use case for `unknown` over `any`: https://stackoverflow.com/questions/51439843/unknown-vs-any

Optional since i think it will require a decent bit of refactoring here.